packages/material-ui/src/TextareaAutosize/TextareaAutosize.d.ts

```diff
@@ -4,6 +4,10 @@ export interface TextareaAutosizeProps extends React.TextareaHTMLAttributes<HTML
   rowsMax?: string | number;
 }
 
-declare const TextareaAutosize: React.ComponentType<TextareaAutosizeProps>;
+declare const TextareaAutosize: React.ComponentType<
+  TextareaAutosizeProps & {
+    ref?: TextareaAutosizeProps extends { ref?: infer RefType } ? RefType : React.Ref<unknown>;
```

Comment:
```suggestion
    ref?: React.Ref<HTMLTextAreaElement>;
```

Would work as well. We don't need to infer a type we declare 3 lines above.